spring-integration-http/src/test/java/org/springframework/integration/http/config/HttpOutboundChannelAdapterParserTests.java

```diff
@@ -166,12 +178,42 @@ public void fullConfig() {
 	}
 
 	@Test
+	public void asyncMinimalConfig() {
+		DirectFieldAccessor endpointAccessor = new DirectFieldAccessor(this.asyncMinimalConfig);
+		AsyncRestTemplate asyncRestTemplate =
+				TestUtils.getPropertyValue(this.asyncMinimalConfig, "handler.asyncRestTemplate", AsyncRestTemplate.class);
+		assertNotSame(this.asyncRestTemplate, asyncRestTemplate);
+		AsyncHttpRequestExecutingMessageHandler handler = (AsyncHttpRequestExecutingMessageHandler) endpointAccessor.getPropertyValue("handler");
+		DirectFieldAccessor handlerAccessor = new DirectFieldAccessor(handler);
+		assertEquals(false, handlerAccessor.getPropertyValue("expectReply"));
+		assertEquals(this.applicationContext.getBean("requests"), endpointAccessor.getPropertyValue("inputChannel"));
+		assertNull(handlerAccessor.getPropertyValue("outputChannel"));
+		DirectFieldAccessor templateAccessor = new DirectFieldAccessor(handlerAccessor.getPropertyValue("asyncRestTemplate"));
+		AsyncClientHttpRequestFactory asyncRequestFactory = (AsyncClientHttpRequestFactory)
+				templateAccessor.getPropertyValue("asyncRequestFactory");
+		assertTrue(asyncRequestFactory instanceof SimpleClientHttpRequestFactory);
+		Expression uriExpression = (Expression) handlerAccessor.getPropertyValue("uriExpression");
+		assertEquals("http://localhost/test1", uriExpression.getValue());
+		assertEquals(HttpMethod.POST.name(), TestUtils.getPropertyValue(handler, "httpMethodExpression", Expression.class).getExpressionString());
+		assertEquals(Charset.forName("UTF-8"), handlerAccessor.getPropertyValue("charset"));
+		assertEquals(true, handlerAccessor.getPropertyValue("extractPayload"));
+	}
+
+	@Test
 	public void restTemplateConfig() {
 		RestTemplate restTemplate =
 			TestUtils.getPropertyValue(this.restTemplateConfig, "handler.restTemplate", RestTemplate.class);
 		assertEquals(customRestTemplate, restTemplate);
 	}
 
+	@Test
+	public void asyncRestTemplateConfig() {
+		AsyncRestTemplate asyncRestTemplate = TestUtils.getPropertyValue(
+				this.asyncRestTemplateConfig,
+				"handler.asyncRestTemplate", AsyncRestTemplate.class);
+		assertEquals(this.asyncRestTemplate, asyncRestTemplate);
```

Comment:
M-m-m. `assertSame()` would be better to be sure that we test against the same instance as it is by our config.